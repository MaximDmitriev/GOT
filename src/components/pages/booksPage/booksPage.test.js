import React from 'react';
import BooksPage from "./booksPage";
import {shallow} from "enzyme";

describe("Testing <BooksPage />", () => {
    const booksPage = shallow(<BooksPage />);

    it("BooksPage have rendered", () => {
        expect(booksPage).toMatchSnapshot();
    });
    
    it("BooksPage state 'error' is false", () => {
        expect(booksPage.state().error).toBeFalsy();
    });

    it("Testing componentDidCatch", () => {
        booksPage.instance().componentDidCatch();
        expect(booksPage.state().error).toBeTrue();
    });
});