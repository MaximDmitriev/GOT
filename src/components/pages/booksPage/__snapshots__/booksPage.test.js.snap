// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <BooksPage /> BooksPage have rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BooksPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Col
        md={
          Object {
            "offset": 3,
            "size": 3,
          }
        }
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />
      </Col>,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "md": Object {
          "offset": 3,
          "size": 3,
        },
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getData": [Function],
          "onItemSelected": [Function],
          "renderItem": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          md={
            Object {
              "offset": 3,
              "size": 3,
            }
          }
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <ItemList
            getData={[Function]}
            onItemSelected={[Function]}
            renderItem={[Function]}
          />
        </Col>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ItemList
            getData={[Function]}
            onItemSelected={[Function]}
            renderItem={[Function]}
          />,
          "md": Object {
            "offset": 3,
            "size": 3,
          },
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "getData": [Function],
            "onItemSelected": [Function],
            "renderItem": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
